feat(timeseries): This adds an incomplete key to the timeseries

wmak opened the pull request:
- This matches how the frontend decides which buckets are still incomplete by marking buckets that occurred in the last 90 seconds

wmak committed:
feat(timeseries): This adds an incomplete key to the timeseries

- This matches how the frontend decides which buckets are still
  incomplete by marking buckets that occurred in the last 90 seconds
- TODO: need to update other tests still

wmak committed:
fix: tests

gggritso commented on src/sentry/api/endpoints/organization_events_timeseries.py:
```diff
@@ -377,3 +395,9 @@ def serialize_timeseries(self, result: SnubaTSResult, axis: str, rollup: int) ->
 
         return timeseries
+
+    def check_incomplete(self, now: float, row: dict[str, Any]) -> str | None:
+        if now - row["time"] < INGESTION_DELAY:
+            return INGESTION_DELAY_MESSAGE
+        else:
+            return None
```
👀 this is a nit, but when we do this on the frontend, we also take the bucket size into account

e.g., if if the current time is 1000, the current row time is 900, the bucket size is 100, and the delay is 90, we mark the last bucket as incomplete because the current time in within the bucket. Would be nice to have that here, too!

wmak replied:
copied the code from FE 👍

gggritso reviewed (commented):
Makes sense, thanks! Small nit about which points are in the incomplete bucket

wmak committed:
ref: address PR comments

- Include bucket size when determining whether incomplete
- Change message to something for the FE to check so we can translate

gggritso reviewed (approved):
Awesome!
